### ISSUE TITLE
How to manage very long string patterns

### ISSUE THREAD

**Issue by jessegreenberg** [phetsims/balloons-and-static-electricity#329]
Form #328, For a11y @zepumph noticed that this sim is using a function called `stripPlaceholders` to manage very long string patterns such as 
```
grabbedFullPatternString: 'Grabbed. {{location}} {{balloonCharge}} {{inducedCharge}} {{positiveCharge}} {{objectCharge}} {{help}}'
```

Depending on sim state, some information needs to be removed.  I need a string pattern for 
- `'Grabbed. {{location}} {{balloonCharge}} {{inducedCharge}} {{positiveCharge}} {{objectCharge}}',`
- `'Grabbed. {{location}} {{balloonCharge}} {{inducedCharge}} {{objectCharge}}',`
- `'Grabbed. {{location}} {{balloonCharge}} {{objectCharge}}',`
- `'Grabbed. {{location}} {{balloonCharge}} {{inducedCharge}}',`
- `'Grabbed. {{location}} {{balloonCharge}}',`
- `'Grabbed. {{location}}',`

Traditionally, we would have 6 string patterns for this. I have been using the full `grabbedFullPatternString`, removing what I don't need with 

```
var patternString = BASEA11yStrings.stripPlaceholders( grabbedFullPatternString, [ 'positiveCharge', 'objectCharge' ] );
```

If this is OK, it should move to StringUtils. But it could be problematic for i18n. Or maybe there is a different way to manage these patterns.

I think the first thing to do is think about how this could impact i18n.

**Comment by jessegreenberg** [comment]
One downside of this for i18n is that  translator cannot change order of alerts depending on context. Maybe a translator would want to have

```
'Grabbed. {{location}} {{balloonCharge}} {{inducedCharge}} {{positiveCharge}} {{objectCharge}}',
'Grabbed. {{location}} {{inducedCharge}} {{balloonCharge}} {{objectCharge}}',
'Grabbed. {{balloonCharge}} {{location}} {{objectCharge}}',
```

Which isn't supported by `stripPlaceholders`

**Comment by jessegreenberg** [comment]
I think the most i18n friendly (and probably the most maintainable) way to do this would be to build of larger string patterns from smaller composite string patterns. Rather than the list of possibilities in the original ticket, we would build up strings using one of these

`Grabbed. {{location}}`.
'Grabbed. {{location}} {{charge}}.

Then filling in charge with one of
'{{balloonCharge}} {{inducedCharge}} {{positiveCharge}} {{objectCharge}}'
'{{inducedCharge}} {{positiveCharge}} {{objectCharge}}'
'{{inducedCharge}} {{objectCharge}}'
'{{objectCharge}}'

Cases will be more clear and we won't have to modify the translatable string, which doesn't seem acceptable.

**Comment by jessegreenberg** [comment]
stripPlaceholders is only used 7 times, that is good news.

**Comment by jessegreenberg** [comment]
stripPlaceholders has been removed.

**Comment by jessegreenberg** [comment]
Last, we should remove `fragmentToSentance`, which has 5 usages in BalloonDescriber.

**Comment by jessegreenberg** [comment]
Much better. Closing.
